src/main/java/mil/dds/anet/search/AbstractSearchQueryBuilder.java

```diff
@@ -143,11 +143,24 @@ public void addBatchClause(AbstractBatchParams<B, T> batchParams,
     batchParams.addQuery(outerQb, this);
   }
 
-  public final void addDateClause(String paramName, String fieldName, Comparison comp,
-      Instant fieldValue) {
-    if (fieldValue != null) {
-      whereClauses.add(String.format("%s %s :%s", fieldName, comp.getOperator(), paramName));
-      DaoUtils.addInstantAsLocalDateTime(sqlArgs, paramName, fieldValue);
+  public final void addDateRangeClause(String startParamName, String startFieldName,
+      Comparison startComp, Instant startFieldValue, String endParamName, String endFieldName,
+      Comparison endComp, Instant endFieldValue) {
+    if (startFieldValue != null) {
+      whereClauses
+          .add(String.format("%s %s :%s", startFieldName, startComp.getOperator(), startParamName));
+      DaoUtils.addInstantAsLocalDateTime(sqlArgs, startParamName, startFieldValue);
+      if (endFieldValue == null && DaoUtils.isRelativeDate(startFieldValue)
```

Comment:
Rewrite this as an `else` to the `if` below (and make `DaoUtils.isRelativeDate(null)` return `false`).